### DISCUSSION TITLE
Generated plugin's api doesn't contains default crud controllers

### DISCUSSION THREAD

**Discussion by daksamit** [strapi/strapi#8013] [Legacy Topics]
I wonder how to enable basic crud routes and controllers like enabled by default while generating basic api.
Only visible one is index in root pugin controller.
```
  index: async (ctx) => {
    // Add your own logic here.

    // Send 200 `ok`
    ctx.send({
      message: "ok",
    });
  },
```
it is possible to inherit `find`, `findeOne`, `create`, `update` and `delete` controllers per every single api in custom plugin?

[core controllers](https://strapi.io/documentation/v3.x/concepts/controllers.html#core-controllers) and [core services](https://strapi.io/documentation/v3.x/concepts/controllers.html#core-controllers) docs aren't clear and don't explain how to deal with custom plugins.

There is also a find service but only for strapi (restaurant) API.
```
const { sanitizeEntity } = require('strapi-utils');

module.exports = {
  /**
   * Retrieve records.
   *
   * @return {Array}
   */

  async find(ctx) {
    let entities;
    if (ctx.query._q) {
      entities = await strapi.services.restaurant.search(ctx.query);
    } else {
      entities = await strapi.services.restaurant.find(ctx.query);
    }

    return entities.map(entity => sanitizeEntity(entity, { model: strapi.models.restaurant }));
  },
};
```

Should I use `strapi.services`, `strapi.entityService` or `strapi.query(model, plugin)` ?
What is the best approach for that case?

Or maybe add option to `routes.json` to use core service for example?

```
{
  "method": "POST",
  "path": "/restaurants",
  "handler": "restaurant.create",
  "core": "create",
  "config": {
    "policies": []
  }
}
```

**Comment by derrickmehaffy** [comment]
So by default models generated in the core ( the `./api/*` folder ) will use the core-api by default without the controllers and services being defined. The main goal here was to make it easier during updates without having to manually update all your models.

You can find a breakdown of the core-api here: https://github.com/strapi/strapi/tree/master/packages/strapi/lib/core-api

Custom plugins however generally are geared towards very custom actions and purpose built controllers and services, so we don't attach them to the core-api (mainly to prevent bloat) but by all means you can just as easily pull the core controllers and services from our docs that you referenced.

For the moment we don't play on adding the core-api to plugins. But I'll ping @alexandrebodin just in case he has anything to add.

**Reply by daksamit** [reply]
Thanks for your clarification. It totally makes sense.
I think it really helped :)
